sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -322,11 +322,6 @@ object SQLConf {
       .intConf
       .createWithDefault(4000)
 
-  val PARTITION_DISCOVERY_ENABLED = SQLConfigBuilder("spark.sql.sources.partitionDiscovery.enabled")
```

Comment:
PR https://github.com/apache/spark/pull/11509 removes the usage of `PARTITION_DISCOVERY_ENABLED`. I think it is safe to remove this conf now.